app/controllers/register_controller.rb

```diff
@@ -5,20 +5,25 @@ def new
 
   def create
     user = User.create(user_params)
-    if user.save
-      flash[:notice] = "Welcome #{params[:name]} Registration Successful"
-      session[:user_id] = user.id
-      redirect_to "/profile"
-    else
-      flash[:error] = user.errors.full_messages.to_sentence
-      @user_info = user_params
-      render :new
-    end
+    user.save ? success(user) : failure(user)
```

Comment:
If anyone finds the ternary difficult for readability, I'm open to changing this to a traditional `if else` statement. Either way, I think breaking out the logic into helper methods really cleans up the code and then adheres to Sandi Metz' Rules For Developers.